### PR TITLE
Migration: User edit

### LINKED ISSUE #23205: RadioButtonGroup: Conflict when two different button groups have the same value on the same page
Reproduced in https://github.com/grafana/grafana/pull/23110#issuecomment-606052699
I think it has to do with conflicting id's https://github.com/grafana/grafana/blob/aacdfcd1a057b4ef49e115642a793fb0420fd4b0/packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.tsx#L78

### PR DESCRIPTION
**Pull request by tskarhed**
#22862

Created an OrgRolePicker, since I've seen a similar use in many places.

I made "Remove from organisation" destructive, but I do think it's too hard on the eyes. Should I make it the `link` variant or use a trash can icon instead? Or maybe used the DeleteButton component.
Screenshot from 2020-03-26 14-38-36

Fixes #23205

### CODE REVIEW

File: public/app/features/admin/UserOrgs.tsx [hunk 1]
```diff
@@ -1,16 +1,17 @@
 import React, { PureComponent } from 'react';
 import { css, cx } from 'emotion';
-import { Modal, Themeable, stylesFactory, withTheme, ConfirmButton, Button } from '@grafana/ui';
+import { Modal, Themeable, stylesFactory, withTheme, ConfirmButton, Button, DeleteButton } from '@grafana/ui';
```
Comment (tskarhed) [inline public/app/features/admin/UserOrgs.tsx]:
```suggestion
import { Modal, Themeable, stylesFactory, withTheme, ConfirmButton, Button } from '@grafana/ui';
```

**Comment by tskarhed** [comment]
Added a remove icon instead of the text.

File: public/app/core/components/Select/OrgPicker.tsx [hunk 2]
```diff
@@ -54,18 +54,16 @@ export class OrgPicker extends PureComponent<Props, State> {
...
-      </div>
+      <Forms.AsyncSelect
+        className={className}
+        isLoading={isLoading}
+        defaultOptions={true}
+        isSearchable={false}
+        loadOptions={this.getOrgOptions}
+        onChange={onSelected}
+        placeholder="Select organization"
+        noOptionsMessage="No organizations found"
```
Comment (hugohaggmark) [inline public/app/core/components/Select/OrgPicker.tsx]: nit: this was a function before, what is the correct prop? Or do we support both?

File: public/app/features/admin/UserPermissions.tsx [hunk 3]
```diff
@@ -57,7 +62,12 @@ export class UserPermissions extends PureComponent<Props, State> {
...
                     <td colSpan={2}>
                       <div className="gf-form-select-wrapper width-8">
-                        <select
+                        <RadioButtonGroup
+                          options={adminOptions}
+                          value={currentAdminOption}
+                          onChange={this.onAdminOptionSelect}
+                        />
+                        {/* <select
                           value={currentAdminOption}
```
Comment (alexanderzobnin) [inline public/app/features/admin/UserPermissions.tsx]: I probably forget to delete this commented block last time I worked on this :)

**Review by alexanderzobnin — APPROVED** [review]
Looks good! Just a minor thing - delete commented block I forget to delete.

File: public/app/features/admin/UserOrgs.tsx [hunk 4]
```diff
@@ -174,7 +164,7 @@ class UnThemedOrgRow extends PureComponent<OrgRowProps, OrgRowState> {
                 onCancel={this.onCancelClick}
                 onConfirm={this.onOrgRemove}
               >
-                Remove from organisation
```
Comment (alexanderzobnin) [inline public/app/features/admin/UserOrgs.tsx]: We decided to have text here when have been working on page design with @sarlinska

**Review by alexanderzobnin — CHANGES_REQUESTED** [review]

File: public/app/features/admin/UserOrgs.tsx (hunk 4 above)
Comment (tskarhed) [reply public/app/features/admin/UserOrgs.tsx]: Gotcha, I'll just use the link version of the button, so it isn't so much `INYOURFACE`

File: public/app/core/components/Select/OrgPicker.tsx (hunk 2 above)
Comment (tskarhed) [reply public/app/core/components/Select/OrgPicker.tsx]: With the new Select it is supposed to be a string. No reason why it was a function before.

**Comment by tskarhed** [comment]
I found a bug
1.  Click "Change role"  on any organisation
Screenshot from 2020-03-30 16-58-00

2. Click "Add user to organization"
3. Try to change role there
Screenshot from 2020-03-30 16-58-08

**Comment by dprokop** [comment]
Found some minor UI glitch: https://youtu.be/hbzl5tf9F74

File: public/app/features/admin/UserOrgs.tsx [hunk 5]
```diff
@@ -241,20 +231,12 @@ export class AddToOrgModal extends PureComponent<AddToOrgModalProps, AddToOrgMod
       <Modal className={styles.modal} title="Add to an organization" isOpen={isOpen} onDismiss={this.onCancel}>
         <div className="gf-form-group">
           <h6 className="">Organisation</h6>
```
Comment (dprokop) [inline public/app/features/admin/UserOrgs.tsx]: Any chance we replace these with Field component please? :)

File: public/app/features/admin/UserOrgs.tsx [hunk 6]
```diff
@@ -241,20 +231,12 @@ export class AddToOrgModal extends PureComponent<AddToOrgModalProps, AddToOrgMod
...
-          <OrgPicker className="width-25" onSelected={this.onOrgSelect} />
+          <OrgPicker onSelected={this.onOrgSelect} />
         </div>
         <div className="gf-form-group">
           <h6 className="">Role</h6>
           <div className="gf-form-select-wrapper width-16">
```
Comment (dprokop) [inline public/app/features/admin/UserOrgs.tsx]: Same as comment above ^^

File: public/app/features/admin/UserProfile.tsx [hunk 7]
```diff
@@ -265,13 +265,13 @@ export class UserProfileRow extends PureComponent<UserProfileRowProps, UserProfi
...
-            <Input
-              className="width-20"
+            <Forms.Input
+              size="md"
               type={inputType}
               defaultValue={value}
               onBlur={this.onInputBlur}
               onChange={this.onInputChange}
-              inputRef={this.setInputElem}
+              ref={this.setInputElem}
```
Comment (dprokop) [inline public/app/features/admin/UserProfile.tsx]: So, this should be not necessary for focusing the input, as the input is rendered when the state is `editing` and you can use autofocus property

**Review by dprokop — APPROVED** [review]
Some nits here and there. Other than that looks good

**Review by alexanderzobnin — APPROVED** [review]

**Pull request merged**